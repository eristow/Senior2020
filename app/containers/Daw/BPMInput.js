import React from 'react';
import { connect } from 'react-redux';
import { createStructuredSelector } from 'reselect';
import { compose } from 'redux';
import styled from 'styled-components';
import PropTypes from 'prop-types';

import { useInjectReducer } from 'utils/injectReducer';
import { makeSelectBpm } from './selectors';
import { changeBpm } from './actions';
import reducer from './reducer';

const BPM = styled.input`
  font-family: 'Helvetica Neue', Helvetica, Arial, sans-serif;
  font-size: 18px;
  color: #25ccf7;
  border: 2px solid #25ccf7;
  background: #eee;
  border-radius: 2;
  align-self: center;
  padding: 10px;
  margin: 2px 5px;
  width: 75px;
`;

const key = 'daw';

export function BPMInput({ bpm, setBpm }) {
  useInjectReducer({ key, reducer });
  return <BPM type="number" value={bpm} min="50" max="180" onChange={setBpm} />;
}

BPMInput.propTypes = {
  bpm: PropTypes.string,
  setBpm: PropTypes.func,
};

const mapStateToProps = createStructuredSelector({
  bpm: makeSelectBpm(),
});

const mapDispatchToProps = dispatch => ({
  setBpm: evt => {
    dispatch(changeBpm(evt.target.value));
  },
});

const withConnect = connect(
  mapStateToProps,
  mapDispatchToProps,
);

export default compose(withConnect)(BPMInput);
