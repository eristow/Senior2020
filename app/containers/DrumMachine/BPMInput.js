import React from 'react';
import { connect } from 'react-redux';
import { createStructuredSelector } from 'reselect';
import { compose } from 'redux';
import styled from 'styled-components';
import PropTypes from 'prop-types';

import { useInjectReducer } from 'utils/injectReducer';
import { makeSelectBpm } from './selectors';
import { changeBpm } from './actions';
import reducer from './reducer';

const BPM = styled.input`
  color: #25ccf7;
  border: 2px solid #25ccf7;
  font-size: 18px;
  background: #eee;
  padding: 10px;
  font-family: 'Helvetica Neue', Helvetica, Arial, sans-serif;
  border-radius: 2;
  margin: 2px 4px;
  margin-right: 20px;
  align-self: center;
`;

const key = 'drumMachine';

export function BPMInput({ bpm, setBpm }) {
  useInjectReducer({ key, reducer });
  return <BPM type="number" value={bpm} min="50" max="180" onChange={setBpm} />;
}

BPMInput.propTypes = {
  bpm: PropTypes.string,
  setBpm: PropTypes.func,
};

const mapStateToProps = createStructuredSelector({
  bpm: makeSelectBpm(),
});

const mapDispatchToProps = dispatch => ({
  setBpm: evt => {
    dispatch(changeBpm(evt.target.value));
  },
});

const withConnect = connect(
  mapStateToProps,
  mapDispatchToProps,
);

export default compose(withConnect)(BPMInput);
