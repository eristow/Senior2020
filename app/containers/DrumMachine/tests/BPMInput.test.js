import React from 'react';
import { render } from 'react-testing-library';
import { Provider } from 'react-redux';
import configureStore from '../../../configureStore';

import { BPMInput } from '../BPMInput';

describe('<BPMInput />', () => {
  let store;

  beforeAll(() => {
    store = configureStore();
  });

  it('Expect to not log errors in console', () => {
    const spy = jest.spyOn(global.console, 'error');
    const dispatch = jest.fn();

    render(
      <Provider store={store}>
        <BPMInput dispatch={dispatch} />
      </Provider>,
    );
    expect(spy).not.toHaveBeenCalled();
  });

  it('Should render and match the snapshot', () => {
    const {
      container: { firstChild },
    } = render(
      <Provider store={store}>
        <BPMInput />
      </Provider>,
    );
    expect(firstChild).toMatchSnapshot();
  });
});
