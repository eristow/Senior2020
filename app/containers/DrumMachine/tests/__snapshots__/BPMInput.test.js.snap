// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<BPMInput /> Should render and match the snapshot 1`] = `
<input
  class="BPMInput__BPM-sc-83ibx4-0 jqkdjn"
  max="180"
  min="50"
  type="number"
  value=""
/>
`;
